### ISSUE TITLE
Use `@bcrypt` in input type

### ISSUE THREAD

**Issue by winter-ice** [nuwave/lighthouse#372]
**Is your feature request related to a problem? Please describe.**

I want to use  `@bcrypt`  in `Input` type.

**Describe the solution you'd like**

Currently,We can only use `@bcrypt`  in `Mutation`,Like this

```graphql
type Mutation {
    createUser(name:String!,email:String!,password:String! @bcrypt): User @create
}
```

This is the way i want:

```graphql
input CreateUser{
    name: String!
    email:String!
    password: String! @bcrypt
}

type Mutation {
    createUser(input:CreateUser!): User @create(flatten:true)
}
```

**Comment by spawnia** [comment]
Yes, this is currently not possible. Unfortunately, it is not trivial to implement - accepting PR's!

**Comment by yaquawa** [comment]
@spawnia How about something like this?

```php
namespace Nuwave\Lighthouse\Schema\Directives\Args;

use Nuwave\Lighthouse\Schema\Values\ArgumentValue;
use Nuwave\Lighthouse\Schema\Directives\BaseDirective;
use Nuwave\Lighthouse\Schema\Values\FieldValue;
use Nuwave\Lighthouse\Support\Contracts\ArgMiddleware;

class BcryptDirective extends BaseDirective implements ArgMiddleware, InputObjectFieldTransformer
{
    /**
     * Directive name.
     *
     * @return string
     */
    public function name(): string
    {
        return 'bcrypt';
    }

    /**
     * Apply transformations on the ArgumentValue.
     *
     * @param ArgumentValue $argumentValue
     * @param \Closure       $next
     *
     * @return ArgumentValue
     */
    public function handleArgument(ArgumentValue $argumentValue, \Closure $next): ArgumentValue
    {
        $argumentValue->addTransformer(
            function ($password) {
                return bcrypt($password);
            }
        );

        return $next($argumentValue);
    }

    public function handleTransformation(FieldValue $fieldValue, \Closure $next): FieldValue
    {
        $fieldValue->addTransformer(
            function (FieldValue $fieldValue) {
                // modify the value of the files
                return $fieldValue;
            }
        );

        return $next($fieldValue);
    }

}
```

**Comment by yaquawa** [comment]
@spawnia Thinking of how to fix this…

**Comment by spawnia** [comment]
No need to have a seperate interface. InputObjectFields and FieldArguments are both of the same underyling AST type, `InputValueDefinitionNode`.

A problem with the current approach is that `ArgumentValue` always expects to have a parent that is a field, which is simply not true when you consider input objects. This is closely related to https://github.com/nuwave/lighthouse/issues/184

**Comment by yaquawa** [comment]
@spawnia Good to know you are planning to rework the architecture of directives!! I'm really want to see how the final implementation of directives could be!

**Comment by yaquawa** [comment]
@spawnia I guess I should just hotfix this issue directly in the resolver until your final implementation finished.💖

**Comment by yaquawa** [comment]
@chrissm79 This can be closed due to #441.

**Comment by chrissm79** [comment]
Thanks @yaquawa!
